src/rust/zh.manhuagui/src/decoder.rs

```diff
@@ -0,0 +1,418 @@
+// use core::{iter::Map, ops::{Index, IndexMut}};
+
+use aidoku::{
+	prelude::format,
+	std::{String, Vec},
+};
+use alloc::vec;
+
+pub struct Decoder {
+	func: String,
+	a: i32,
+	c: i32,
+	data: Vec<String>,
+}
+
+impl Decoder {
+	pub fn new(document: String) -> Self {
+
+		let script = get_script(document);
+
+		let func = get_func(script.clone());
+		let a = get_a(script.clone(), func.clone()).parse::<i32>().unwrap_or(-1);
+		let c = get_c(script.clone(), func.clone()).parse::<i32>().unwrap_or(-1);
+		let data: Vec<String> = get_data(script, func.clone());
+
+		Decoder { func, a, c, data }
+	}
+
+	fn e(&self, c: i32) -> String {
+		let prefix: String = if c >= self.a { self.e(c / self.a) } else { String::new() };
+
+		let suffix_vec = vec![
+			self.tr(c % self.a, 36),
+			String::from_utf8(vec![(c % self.a + 29) as u8]).unwrap(),
```

Comment:
you're still unwrapping here, provide a default value
```suggestion
			String::from_utf8(vec![(c % self.a + 29) as u8]).unwrap_or_default(),
```